### ISSUE TITLE
[18.0] sale/purchase/account: Not able to replace the product display name with custom text

### ISSUE THREAD

**Issue by pedrobaeza** [odoo/odoo#189870]
### Odoo Version

- [ ] 16.0
- [ ] 17.0
- [X] 18.0
- [ ] Other (specify)

### Steps to Reproduce

**Steps to reproduce**

- Go to Sales > Quotations
- Create a new one.
- Add a line and select "Customizable Desk". You can leave the defaults for selecting the variant.
- You get this:

  ![imagen](https://github.com/user-attachments/assets/5c633300-5f59-47f6-b4ce-5fa1e6f8ae51)

**Current behavior**

You can modify the text in italics, but there's no way to remove nor customize the product displayed name (`[FURN_0096] Customizable Desk (Steel, White)`). And this printed in the reports.

**Expected behavior**

Be able to change whole text.

There are a lot of valid use cases for not wanting that text to be displayed:

- You use a generic product (for example, `Hour of development`), and in each quotation, you write down the development itself, but you don't want to who in the first line the text "Hour of development".
- You have your own internal references, but your customer uses another references. Then, on the quotations to that customer, you put their reference, not yours. Right now, you internal reference will always be displayed, although you put the other reference. As reference, we have a module in OCA for directly doing the switch of the reference from master data stored in the product (`product_supplierinfo_for_customer_sale` - the module name is an inherited and long history, hehe - ).
- Other business reasons to obfuscate your internal product name.

**Possible design solution**

I was thinking in adding an integrated "sonoff" button to enable/disable the display of the product name:

![imagen](https://github.com/user-attachments/assets/21127db0-78b5-4517-b5ea-c399cc3d62b4)

**Comment by pedrobaeza** [comment]
@Feyensv I would like to have your opinion on this.

**Comment by Feyensv** [comment]
Little tip, you can hide the 'Product' column to be able to update the full line description
![image](https://github.com/user-attachments/assets/42fdd0b6-4897-42ec-be30-e541ec76870e)

About your point, I agree that despite being quite nice to reduce the space taken by the product & description columns, this new widget restrains the freedom of users to update the full line description.

Your idea might be a way to solve it, I'll forward the feedback, but I doubt we'll change anything for now (even less in stable unless a lot of people complain about it).

**Comment by pedrobaeza** [comment]
Thanks for the tip, Victor. Yeah, it works, but it requires to do the hiding each time, and to not touch again the description with the "Product" column shown, or it will incorporate it again. Let's see if this feedback is considered to be changed.

**Comment by yanghongeer** [comment]
yes. the same proble. dont want to show product name in pdf report .just show description.
directly show whole description below product column may be better

**Comment by yanghongeer** [comment]
Especially when the product is in multiple languages, the PDF will display the basic language product name, customer language product name, and description, which is terrible

**Comment by Webbeh** [comment]
Oh my gosh, I've been looking for a fix for this all afternoon and trying to edit the invoice report to not include this, to no avail.

Please let us, indeed, change the behavior so we do not need to display the product name. As I work in service, my invoices are riddled with "Labour xxxxx" "Labour yyyy" "Labour zzzz" text. I do not need to add the "Labour" in the description.

This is a regression from 17 (where it already was painful to have to remove the product name automatically added to the description), where we could completely get rid of this.

Currently, I'm trying to edit the invoice report as I said. Can anyone tell me what t-field I should use if that's even possible ? 
Default is `line.name`, I have tried `line.label`, `line.desc`, `line.description`, `line.product_id.description`, and quite a few others to no avail.

![Image](https://github.com/user-attachments/assets/b7547243-64f1-4767-91df-7883cc34d233)

Thank you.

Note that I have an enterprise licence.

**Comment by ploegvde** [comment]
We have the same issue with this and raised a support ticket. No luck at the moment. This is one of the responses.

It is indeed a regression and blocked multiple migrations to Odoo 18 for our clients. All clients use some sort of generic product and that cannot be used anymore. :(



-----
The problem sadly comes from the implementation of this new widget in accounting, which also impacts sale and purchase orders.
The widget enforces the presence of the product name in the description, which was indeed done by default previously, but customers could always remove the product name from the description manually.

It is possible to remove it from the description if you hide the Product and Product Variant column, before modifying the description, even though it's complex to apply, and the name is reset automatically in some flows.

**Comment by Webbeh** [comment]
The problem when you do remove the "Product" column is that the analytics are broken because of it (the amount of items sold are not incremented, since it's no longer the same item).  In turn, this will break any automatic inventory management and any automatic project/task creation upon sale of a product...

I don't understand how they did their testing for this, if they did any.

Honestly, the only thing that they need to do is hide the name of the product on quotes/invoices, it shouldn't be that hard to do. OP's suggestion is actually quite good, but I would add that there should be a per-product default setting as well.

**Comment by ploegvde** [comment]
If you use a sales template, it works correctly (there are still 2 fields, but maybe not for long). So, you can have 2 of the same products on a quote, one from the sales template and one manual added, that is presented differently on the quote 🤷‍♀️

**Comment by pedrobaeza** [comment]
On v16, we have backported the widget at https://github.com/OCA/web/tree/16.0/web_widget_product_label_section_and_note, and we have added the button I proposed as design for hiding/showing the product name:

![Image](https://github.com/user-attachments/assets/4fdb8486-40c3-4330-a0cc-ad8c0ceeabd3)

You can extract that part and have a module to complement it as part of the 18 core while Odoo is looking for a solution (which probably won't land in 18), and if you share it in OCA, it would be wonderful.

**Comment by Feyensv** [comment]
The issue has been forwarded and will be discussed in a coming meeting between teams.
I can't guarantee any change in the future, but at least this issue and the noticed problems will be considered more attentively.

If you notice any other issue or incoherence related to this new widget (be it for SO/PO or AM), please tell us about it.
Don't hesitate to link any support ticket you open about it too, to centralize the topic here.

**Comment by ploegvde** [comment]
https://www.odoo.com/nl_NL/my/tasks/4485308

**Comment by Webbeh** [comment]
@ploegvde you do know that we cannot access your tickets, right?

**Comment by ploegvde** [comment]
> [@ploegvde](https://github.com/ploegvde) you do know that we cannot access your tickets, right?

Yes I know. But the employees of Odoo can. As asked yesterday

**Comment by ploegvde** [comment]
> The issue has been forwarded and will be discussed in a coming meeting between teams. I can't guarantee any change in the future, but at least this issue and the noticed problems will be considered more attentively.
> 
> If you notice any other issue or incoherence related to this new widget (be it for SO/PO or AM), please tell us about it. Don't hesitate to link any support ticket you open about it too, to centralize the topic here.

Hi, Curious how the meeting wend and if there is good news?

**Comment by Feyensv** [comment]
It's happening next week.
As soon as I have any feedback, I'll post it here ;)

**Comment by Webbeh** [comment]
You can hide the product name and only have its description appear by using `line.product_id.description_sale` in the XML of the invoice report.

Example video :
https://drive.google.com/file/d/1ZkCQLNBVtvIUaR3zrIb-1gtf1Rt6l8n-/view

**Comment by pedrobaeza** [comment]
@Webbeh That's still a fixed value, not a variable one per order.

**Comment by Webbeh** [comment]
Yeah I just tried that and noticed the same. Looking at odoo's code, I do not see any way to get this description easily. This is such a shame.

**Comment by Webbeh** [comment]
> https://www.odoo.com/nl_NL/my/tasks/4485308 opw-4485308

I may as well add mine too then : 
https://www.odoo.com/fr_FR/my/tasks/4531617

**Comment by Webbeh** [comment]
EDIT: Sorry for the multiple posts.

I have now tried with a brand new testing database and it seems to do exactly what we would like : 
- If no description is provided (either in product or quote), show the product name
- If no description is provided in the quote, show the product name + the product description
- If a description is provided in the quote, show only the description

This is consistent with what my Odoo partner has, and I assume this issue has then been fixed for the saas version of Odoo.

So this issue is either : 
- a fix has been implemented after the current enterprise revision
- there's a bug with the 17->18 migration that causes the name to be shown

I'd tend to favour the first option but I cannot find any commit related to this... It would be awesome if someone could find the difference.

For reference, self-hosted enterprise is locked at version 2024/12/21.

**Comment by ploegvde** [comment]
I just tested in our Odoo 18 and on Runbot and indeed it works like Odoo 17. There was a fix, apparently. @Feyensv Can you confirm?

**Comment by Webbeh** [comment]
I have created a new database on my on-prem and I have the same behaviour as the other database (the issue isn't fixed there).
On this new database, there is only the Sales module (and its deps).

If anyone needs a test account, send me a message for details.

**Comment by pedrobaeza** [comment]
Yes, I'm seeing now a change of behavior has been set now, but now setting extra text requires to copy/paste the product name if you want to have it in the quote.

**Comment by Webbeh** [comment]
If you have the description in the product and not on the quote page, it will display both the name and the description

**Comment by Webbeh** [comment]
So I have bad news, apparently they WANT to force us to use the name + description for every item as it is "the default expected behaviour" and are redirecting me to very expensive and unneccesary custom development options if we want to have a custom description option, like in version 17 and prior.

I am extremely unappy with this decision and am definitely considering moving away from this product - at least from the paid version.

For transparency's sake, here's a full transcript (with redacted personal information) down here for everyone to be able to follow the conversation I had with support.


Note that ALL of this goes away if there actually is a way to get the name and description separately from within the template, which there doesn't seem to be any way of doing at this moment...
Something along the lines of  `line.product_name` and `line.product_description`...


# Invoice/quote : not able to hide product name (#4 531 617) 
## February 2nd
### Me

Good morning,

A ticket has been opened a few weeks ago on Github, but I'm experiencing the same issue.

https://github.com/odoo/odoo/issues/189870
I am user "Webbeh" on github.

Basically, I would like to hide the product name on all reports/quotes/invoices if they have a description, but still be able to use the new widget format for name/description, if that makes sense.

I tried to modify the invoice template to display (from Odoo Studio) the item description only, but I cannot find a proper variable containing that info.

The default displayed information in the report is `line.name`.

I have tried line.label, line.desc, line.description, line.product_id.description, and quite a few others to no avail.

Is there any variable I can use to ONLY display the description?

Thank you very much !
Kind regards,

![Image](https://github.com/user-attachments/assets/91ac43b1-859a-45be-bf77-aeeb5230cafb)


## February 3rd
### Odoo
Hello,

Thank you for contacting Odoo Support!

I reviewed your concern and would like to inform you that your requirement is definitely achievable through Odoo Studio. To only print the description of the product and not the name, you can just replace the 'line.name' with 'line.product_id.description_sale' on the XML of the pdf report (as highlighted on your attached screenshot).

Also, please refer the below given video link of the test case demonstrating the same:

https://drive.google.com/file/d/1ZkCQLNBVtvIUaR3zrIb-1gtf1Rt6l8n-/view?usp=sharing

NOTE: It is advised to try the solution on a duplicate of your database to make sure that everything is working fine before you apply the fix to your production database.

Hope you find this helpful. Please let me know in case of any doubt related to this ticket if any.

Thank you.
Have a nice day:)

## February 20th
### Me
Hi again.

This will not work as the "description_sale" comes for the article description and not from the current line's description.

In the attached example, the "Labour" article is there twice with two different descriptions. The article itself has no description and will show "Bacon burger" when I print the quote.

![Image](https://github.com/user-attachments/assets/80fbe9fe-5172-415e-94f4-4be18e8cc93a)


## February 24th
### 12h57 - Odoo
Hello,

Thank you again for reaching out to me.

Yes, you are right. The 'description_sale' will fetch from the product template. If you have 2 lines of same product, system will print the same description from the product template.

Currently, the system displays the product description and hides the product name if the description is available. If no description is provided, the product name is displayed instead. This can be helpful in your case. You can add your custom text in the description so that it gets printed on the report.

Could you please clarify your specific requirements? Are you looking for the description to always be printed, the product name, or would you like both to be printed based on certain conditions?

Waiting for your revert.
Kind regards,


### 13h05 - Me
The issue I'm having is that on quotes and invoices, both the name and the description are displayed. I want to only print the description if one is provided, never print the name unless no description is available.

In the examples provided,
- the first article has NO description (neither in the product, nor in the quote). It has a translation but that's it
- the second article has a description in the QUOTE only (no description in the product)
- the third article has a description in the PRODUCT only (no description in the quote)

What I want to be printed in the quotes/invoices, in order :
- the (translated) name of the article (`Main d'oeuvre`)
- The quote description only (not the name, not the product description if any)
- The product description only (not the name)
So to recap, I only want the "description" line to be printed if a description is provided. Currently, both name and description are printed.

Currently, the name of the article is printed by default and there's nothing we can do about it. The only one behaving correctly is the first item.

EDIT: on the provided screenshots, I'm showing the default Odoo 18 template.
The only way for me to only display the description is to disable the "Product" column in the quote (see last screenshot), but if I do this, I lose analytics on every product....

![Image](https://github.com/user-attachments/assets/7ed94a21-5a06-4ffd-ae5b-307e31d844d3)
![Image](https://github.com/user-attachments/assets/12b5d830-fd82-43fe-9363-67030a36a4d4)
![Image](https://github.com/user-attachments/assets/4966271a-7d35-4d91-8026-7161aedab4a4)



### 13h19 - Odoo
Hello,

Thank you again for your response.

I analyzed your concern in standard runbot and would like to inform you that the point 1 and 2 are by default available in version 18.

If there is no description on the quotation, the product name will get printed.

If you provide description on quotation, only the description will get printed and not the product name.

Regarding your concern 3, it is not possible to display the description from the product template as per your requirements.

Please refer the test case video of version 18 runbot in the below given link:

https://drive.google.com/file/d/1u0bOxcsoqXF_TbPb3vp-qSwi9gJ_lsNt/view?usp=sharing

Do let me know if I misunderstood your concern.

Kind regards,



### 13h27 - Me
This is not what is happening on my install, which is default Odoo (I have no custom studio edits for the sales/quotes/...).

Point 3 I can live with and don't really care. However, for point 2, the name IS getting displayed even if I provide a description in the quote, as you can see in my previous screenshots.

It has been doing so since I upgraded to v18, and the github link I posted shows the same behaviour is happening to other people as well.

https://github.com/odoo/odoo/issues/189870

You may have a more recent build than mine, but since I'm self hosted, there's nothing I can do.

Here's a video proving it.
https://youtu.be/<redacted>


### 13h42 - Odoo
Hello,

Thank you again for your response.

Since your database is on-premise, I would request you to provide me test/staging database along with the login credentials so that I can further investigate the issue for you.

Kind regards,


###15h55 - Me
I have tested on a new database on Odoo.com and it works there. I then made a new database on my on-prem install, and there it doesn't work.

https://<redacted>
User : <redacted>
Pass : <redacted>


## February 26th
### 12h44 - Odoo
Hello,

Thank you for providing the credentials.

I'm currently discussing you case with the internal team. I'll get back to you as soon as I have more information regarding this.

Thanks in advance for your patience.

Kind regards

### 13h53 - Odoo
Hello,

Thank you for your patience.

I've reviewed this internally and confirmed that the system is functioning as expected on your Odoo instance. **However, the issue you're experiencing is a bug in the standard version of Odoo.**

The bug fix team is actively working on this, and once resolved, **the standard version will behave the same way as your instance, where the product name will be displayed along with the description, even if manually added to the quotation.**

In summary, after the bug fix is implemented, the product name will be correctly printed alongside the description when manually added in the quotation same as what it is on your current database.

Hope this information is helpful to you. Do let me know in case of further questions related to this ticket if any.

Thanks and have a great day!!


### 13h55 - Me
Hi...

This is exactly the opposite of what I want....


### 14h03 - Odoo
Hello,

Thank you again for your response.

I understand your concern but majority of the users find this as unintended behavior(a bug) and hence, the bug will get fixed in short time.

Hope you will understand.
Kind regards,


### 14h04 - Me
This ticket will remain open until this issue is fixed or a workaround is found. Since the new widget for name+description we completely lost the ability to do a custom text (and ONLY a custom text) on the quote/invoice. This is a massive regression.


### 14h23 - Odoo
Hello,

Thank you for your feedback but I would like to inform you that the system will function how it should be and convenient to majority of users.

However, you can opt for customization services. For this you would be required to check the possibilities with the consultant, as it is not achieved through Odoo standard practices. You may also internally discuss with your account manager Leia Gutierrez (legu@odoo.com) to have detailed information regarding customization services.

This video explains in short what the success pack is:
https://www.youtube.com/watch?v=R5WHbTHN6qA

I hope this information will help you. Please do let me know if I missed anything or if you have any doubts.

Thank you.
Have a nice day!

**Comment by rvalyi** [comment]
not sure, but it seems this big refactor in master aims at fixing this https://github.com/odoo/odoo/pull/196785

**Comment by Feyensv** [comment]
> I just tested in our Odoo 18 and on Runbot and indeed it works like Odoo 17. There was a fix, apparently. [@Feyensv](https://github.com/Feyensv) Can you confirm?

Indeed, It seems like someone merged a fix that reintroduced the old behavior, but this will be reverted soon, because that's not the behavior requested (and it removed the product name from customer documents, causing another wave of tickets).
For the record: #197788 and incoming revert in #199529

> not sure, but it seems this big refactor in master aims at fixing this

It shouldn't be modifying the behavior at all, it's mostly to clean/remove technical debt.

**Comment by ploegvde** [comment]
Hi @Feyensv What a mess. So, we informed all that it is fixed again, and now it is reverted again? Ohh boy, pfff

I have discussed this with our Odoo partner manager also, and she is surprised. That Odoo does not change it is clear, but we have to explain it to tens of companies. What is Odoo idea/vision of this change? Many companies use a generic product and change the description in the sale order line description. Thanks for the explanation.

**Comment by Feyensv** [comment]
Indeed, this mess clearly highlights how much this feature is unclear, and disturbs existing users.
Nevertheless, the decided 'Odoo' stance is that you are expected to use meaningful product names (or at least names that won't cause any issue/misunderstanding if shown to the customer).

As said before, we still plan to fix the issue where the product name is present twice (translated and untranslated).
I tried to find ways to ease customizations for those that wanted to keep the previous behavior, but wasn't able to find/imagine something really helpful.  Most I could think of was to split the display of the line description in reports/portal, between the first line (with the product name) and the remaining lines (holding the true description), so that the first line could be hidden with a simple xpath, but I doubt it would help much.

**Comment by Webbeh** [comment]
For a service company where labour is the most used product, with TONS of different ways to describe the labour that is never used for another customer again, this change is absolutely stupid.

Again, if we at the very least had the option to do it ourselves in our quotes, it would be great. 

I categorically refuse to go with a custom development route for this kind of work. It is behaviour that existed in prior versions of Odoo and now this new widget breaks the previous behaviour.

**Comment by Bart-dh** [comment]
> Indeed, this mess clearly highlights how much this feature is unclear (even for our employees), and disturbs existing users. Nevertheless, the decided 'Odoo' stance is that you are expected to use meaningful product names (or at least names that won't cause any issue/misunderstanding if shown to the customer).
> 
> As said before, we still plan to fix the issue where the product name is present twice (translated and untranslated). I tried to find ways to ease customizations for those that wanted to keep the previous behavior, but wasn't able to find/imagine something really helpful. Most I could think of was to split the display of the line description in reports/portal, between the first line (with the product name) and the remaining lines (holding the true description), so that the first line could be hidden with a simple xpath, but I doubt it would help much.

IMO, as Odoo partner waiting to upgrade a lot of customers due to this problem, the option to hide or display product names in customer-facing documents would be useful. If working with generic products, product name is not shown and if wanted can be included in the sales description.

Of course just way of working as it is now in <18.0 would be best...

**Comment by Bart-dh** [comment]
As per Odoo's own AI assistent...

<img width="882" alt="Image" src="https://github.com/user-attachments/assets/6ea49e16-3ce0-475b-90c0-f7365f7c352c" />

**Comment by Bart-dh** [comment]
> What's wrong with that AI assistant answer? The question has way to less details and is out of context of this issue...

It recommends reusing products, which is of course not an option if they are planning that product name will always be on the SO and invoice documents.

**Comment by intero-chz** [comment]
> > What's wrong with that AI assistant answer? The question has way to less details and is out of context of this issue...
> 
> It recommends reusing products, which is of course not an option if they are planning that product name will always be on the SO and invoice documents.

You haven't asked about generic products. You have just asked to reuse products. Which ofcourse would be very stupid not to do, for example creating a new Burger product for every burger sold.

**Comment by intero-chz** [comment]
Clarification: I don't like the changes in Odoo 18 either. But it is also not fair to use the AI Assistant with a simple question in such a way that Odoo can be accused of a contradiction here.

**Comment by Webbeh** [comment]
I believe the first thing should be to expose both parts of the computed product name as line variables. 

The data is there, it exists, we just need access to it.  That will help implementing this feature request.

I don't mind if that becomes an OCA module, but I do believe more and more service-based (rather than product-based) users will need this feature.

The argument Odoo gave us that "more people expect this or that behavior" is moot and irrelevant.

This seems like a very little change with very little maintenance attached that will allow an extremely helpful customization. Why not have both options and keep both types of customers happy? Especially since, again, the code already had a "bug" that coild simply be reimplemented as an optional feature...

Let's please stop arguing about whether or not most users want it one way or another. Let's just have both options and never have to argue again...

If I was capable of doing a merge request for this, I would have done it already.

**Comment by brunoc68** [comment]
Dear all, I've just come to this issue while trying to upgrade several databases of customers. Basically we cannot go for v18 because of that. This issue stands full for new customers of course.

The use of a generic product is common and widely used ; this is a feature every software has, including Odoo until V17

Such a widely used feature that just disappeared is not a regression, it is a blocking bug. I expect a correction as soon as possible so that we and our customers can just continue to work.

The correction is by the way very easy : just go back to what it was before.

**Comment by Webbeh** [comment]
As this issue is blocking updates for several customers, I believe it should be marked as high priority and treated as such.

**Comment by pedrobaeza** [comment]
FYI, we have put the feature of allowing to show/hide the product name (which we already had previously in the backport of the whole widget) in a module for 18:

https://github.com/OCA/web/pull/3165

**Comment by Webbeh** [comment]
@pedrobaeza **THANK YOU guys so very damn much!** 
Can't express my gratitude enough!

**Comment by Bart-dh** [comment]
@pedrobaeza thanks, this is indeed amazing. Is there any way to set default value for the visibility?

**Comment by pedrobaeza** [comment]
Sorry, not planned for now.
